core/api/test/integration/app/lightning/delete-ln-payments.spec.ts

```diff
@@ -0,0 +1,100 @@
+import { Lightning } from "@/app"
+
+import { TWO_MONTHS_IN_MS } from "@/config"
+
+import { decodeInvoice } from "@/domain/bitcoin/lightning"
+import { CouldNotFindLnPaymentFromHashError } from "@/domain/errors"
+
+import * as LndImpl from "@/services/lnd"
+import { LnPayment } from "@/services/lnd/schema"
+import { LnPaymentsRepository } from "@/services/mongoose"
+
+const DEFAULT_PUBKEY =
+  "03ca1907342d5d37744cb7038375e1867c24a87564c293157c95b2a9d38dcfb4c2" as Pubkey
+
+const randomRequest =
+  "lnbcrt10n1p39jatkpp5djwv295kunhe5e0e4whj3dcjzwy7cmcxk8cl2a4dquyrp3dqydesdqqcqzpuxqr23ssp56u5m680x7resnvcelmsngc64ljm7g5q9r26zw0qyq5fenuqlcfzq9qyyssqxv4kvltas2qshhmqnjctnqkjpdfzu89e428ga6yk9jsp8rf382f3t03ex4e6x3a4sxkl7ruj6lsfpkuu9u9ee5kgr5zdyj7x2nwdljgq74025p"
+const invoice = decodeInvoice(randomRequest)
+if (invoice instanceof Error) throw invoice
+
+afterEach(async () => {
+  await LnPayment.deleteMany({})
+
+  jest.restoreAllMocks()
+})
+
+describe("deleteLnPaymentsBefore", () => {
+  const lnPayments = LnPaymentsRepository()
+  const { paymentHash, paymentRequest } = invoice
+  const mockedLnPaymentLookup = {
+    createdAt: new Date(Date.now()),
+    paymentHash,
+    paymentRequest,
+  }
+
+  it("persists payment to be deleted in ln-payments collection", async () => {
```

Comment:
How do you mean? The thing I'm trying to check here is of LnPayments.persist gets called inside the use-case

Reply:
https://github.com/GaloyMoney/galoy/pull/3677#discussion_r1425571351

Reply:
How do you mean? There's a code branch in `checkAndDeletePaymentForHash` that persists an lnPayment if it didn't previously exist in LnPaymentsRepository that I explicitly wanted to test

Reply:
I review bottom -> up, there is no need to have 2 tests for this, is just add 
```
const lnPaymentAfter = await lnPayments.findByPaymentHash(paymentHash)
    if (lnPaymentAfter instanceof Error) throw lnPaymentAfter
    expect(lnPaymentAfter.paymentHash).toBe(paymentHash)
```
in second test but it is up to you

Reply:
I'd prefer to have them separate because the 2nd test needs to have a step that marks `isCompleteRecord` as true and I bring that in manually, and if I tried to not do that then the test becomes more complicated because later tests relies on prior state in the test